src/main/java/receive_a_message/SmsHelloWorld.java

```diff
@@ -13,19 +13,65 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 
+import io.github.cdimascio.dotenv.Dotenv;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+
+class IncomingSmsRequest
+{
+    @JsonProperty("accountId")
+    public String accountId;
+    @JsonProperty("applicationId")
+    public String applicationId;
+    @JsonProperty("direction")
+    public String direction;
+    @JsonProperty("from")
+    public String from;
+    @JsonProperty("messageId")
+    public String messageId;
+    @JsonProperty("phoneNumberId")
+    public String phoneNumberId;
+    @JsonProperty("requestType")
+    public String requestType;
+    @JsonProperty("status")
+    public String status;
+    @JsonProperty("text")
+    public String text;
+    @JsonProperty("to")
+    public String to;
+
+    public IncomingSmsRequest() {
+    }
+
+    public IncomingSmsRequest(String accountId, String applicationId, String direction, String from, String messageId, String phoneNumberId, String requestType, String status, String text, String to) {
+        this.accountId = accountId;
+        this.applicationId = applicationId;
+        this.direction = direction;
+        this.from = from;
+        this.messageId = messageId;
+        this.phoneNumberId = phoneNumberId;
+        this.requestType = requestType;
+        this.status = status;
+        this.text = text;
+        this.to = to;
+    }
+}
+
 @RestController
 public class SmsHelloWorld {
 
   @RequestMapping(value = { "/incomingSms" }, method = RequestMethod.POST)
-  public void inboundSms(@RequestBody String body) {
-    String accountId = System.getenv("ACCOUNT_ID");
-    String apiKey = System.getenv("API_KEY");
-    String fromNumber = System.getenv("FREECLIMB_PHONE_NUMBER");
-    String toNumber = System.getenv("TO_PHONE_NUMBER");
+  public void inboundSms(@RequestBody IncomingSmsRequest request) {
+    Dotenv dotenv = Dotenv.configure().ignoreIfMissing().load();
+    String accountId = dotenv.get("ACCOUNT_ID");
+    String apiKey = dotenv.get("API_KEY");
+    String fromNumber = dotenv.get("FREECLIMB_NUMBER");
+    String toNumber = request.from;
```

Comment:
Instead of using request.from, we should have another environment variable for the to variable:

https://github.com/FreeClimbAPI/PHP-SMS-Quickstart/pull/3#discussion_r1585580921